Mathlib/Algebra/BigOperators/Group/Finset/Basic.lean

```diff
@@ -95,53 +98,45 @@ theorem prod_disjUnion (h) :
   rfl
 
 @[to_additive]
-theorem prod_disjiUnion (s : Finset ι) (t : ι → Finset α) (h) :
+theorem prod_disjiUnion (s : Finset κ) (t : κ → Finset ι) (h) :
     ∏ x ∈ s.disjiUnion t h, f x = ∏ i ∈ s, ∏ x ∈ t i, f x := by
   refine Eq.trans ?_ (fold_disjiUnion h)
   dsimp [Finset.prod, Multiset.prod, Multiset.fold, Finset.disjUnion, Finset.fold]
   congr
   exact prod_const_one.symm
 
 @[to_additive]
-theorem prod_union_inter [DecidableEq α] :
+theorem prod_union_inter [DecidableEq ι] :
     (∏ x ∈ s₁ ∪ s₂, f x) * ∏ x ∈ s₁ ∩ s₂, f x = (∏ x ∈ s₁, f x) * ∏ x ∈ s₂, f x :=
   fold_union_inter
 
 @[to_additive]
-theorem prod_union [DecidableEq α] (h : Disjoint s₁ s₂) :
+theorem prod_union [DecidableEq ι] (h : Disjoint s₁ s₂) :
     ∏ x ∈ s₁ ∪ s₂, f x = (∏ x ∈ s₁, f x) * ∏ x ∈ s₂, f x := by
   rw [← prod_union_inter, disjoint_iff_inter_eq_empty.mp h]; exact (mul_one _).symm
 
 @[to_additive]
 theorem prod_filter_mul_prod_filter_not
-    (s : Finset α) (p : α → Prop) [DecidablePred p] [∀ x, Decidable (¬p x)] (f : α → β) :
+    (s : Finset ι) (p : ι → Prop) [DecidablePred p] [∀ x, Decidable (¬p x)] (f : ι → M) :
     (∏ x ∈ s with p x, f x) * ∏ x ∈ s with ¬p x, f x = ∏ x ∈ s, f x := by
-  have := Classical.decEq α
+  have := Classical.decEq ι
   rw [← prod_union (disjoint_filter_filter_neg s s p), filter_union_filter_neg_eq]
 
 @[to_additive]
-lemma prod_filter_not_mul_prod_filter (s : Finset α) (p : α → Prop) [DecidablePred p]
-    [∀ x, Decidable (¬p x)] (f : α → β) :
-    (∏ x ∈ s with ¬p x, f x) * ∏ x ∈ s with p x, f x = ∏ x ∈ s, f x := by
+lemma prod_filter_not_mul_prod_filter (s : Finset ι) (p : ι → Prop) [DecidablePred p]
+    [∀ x, Decidable (¬p x)] (f : ι → M) :
+    (∏ x ∈ s.filter fun x ↦ ¬p x, f x) * ∏ x ∈ s.filter p, f x = ∏ x ∈ s, f x := by
```

Comment:
Is this replacement of the `with` syntax intentional? I expect not.

Reply:
No, accident indeed. Not sure ow that happened though, because that syntax replacement is several months old now